python-threatexchange/threatexchange/hashing/pdq_hasher.py

```diff
@@ -19,7 +19,9 @@ def pdq_from_file(path: pathlib.Path) -> PDQOutput:
     Given a path to a file return the PDQ Hash string in hex.
     Current tested against: jpg
     """
-    img_pil = Image.open(path)
+    # Convert to RGB makes sure the resulting array has the number of dimensions
+    # expected by 'pdqhash' (without this black and white images result in errors).
+    img_pil = Image.open(path).convert("RGB")
```

Comment:
I'm wondering if there's a way to do this conversion only when needed? Dunno if the conversion could ever have a side effect (either now or in the future). Looking at https://pillow.readthedocs.io/en/stable/_modules/PIL/Image.html?highlight=grayscale# there seems to be a way to detect the current state of affairs... if you get L then you'd do this conversion? What do you think?

Reply:
yeah, @eriksendc I went back and forth on that, you can even manually detect the dimensions of the array and add the 3rd one manually. My first solution was adding the following helper:
```
def _check_dimension_and_expand_if_needed(array: np.ndarray) -> np.ndarray:
    if array.ndim == 2:
        array = np.concatenate([array[..., np.newaxis]] * 3, axis=2)
    return array
``` 

The reason I went this route is 1) the output was identical for all the images I tested 2) covert("RGB") will use the library to help us be more error tolerant as the python pdq lib we currently use expects 3 dimensions and I would rather have the images be converted to expected by default.  This way we should handle most cases not just the narrow one of the 'L' check. I'd prefer to avoid similar failures of just not working for some images. Happy to change it back if you feel strongly though :)

Reply:
@BarrettOlson The thing I wonder about is the potential side effect of "converting" something that doesn't really need to be converted in the first place. I understand you tested a particular corpus of images... but without that being a super giant number, what would we really know? Also, the code for this conversion might not have a side effect now, but it could some day. I would recommend a hybrid by the way, detect if array.ndim is 2 and then use `convert()` to do the conversion. I'd say I feel "a little past 50-50" strong about it, but totally good with going with a tie-breaker. @schatten what do you think?

Reply:
@schatten brought up perf and is completely right, this is a bad idea, putting up a change soon :)